cis194/week2/zhansongl/LogAnalysis.hs

```diff
@@ -0,0 +1,63 @@
+module LogAnalysis where
+
+import Log
+
+import Text.Megaparsec hiding (parse)
+import Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer as L
+import Data.Void
+
+type Parser = Parsec Void String
+
+messageTypeParser :: Parser MessageType
+messageTypeParser = do
+  charType <- oneOf "IEW"
+  case charType of
+       'I' -> return Info
```

Comment:
NIT: Because GHC has now added Applicative as a constraint on Monad, we prefer `pure` instead of `return`.

Reply:
yeah, saw this in the wiki after I finished writing this code, will fix